compiler/rustc_middle/src/ty/query.rs

```diff
@@ -114,8 +114,8 @@ macro_rules! query_storage {
     ([storage($ty:ty) $($rest:tt)*][$K:ty, $V:ty]) => {
         <$ty as CacheSelector<$K, $V>>::Cache
     };
-    ([$other:ident $(($($other_args:tt)*))* $(, $($modifiers:tt)*)*][$($args:tt)*]) => {
-        query_storage!([$($($modifiers)*)*][$($args)*])
+    ([$modifier:tt $($rest:tt)*][$($args:tt)*]) => {
```

Comment:
FWIW as noted in the commit message, I don't really fully understand what is going on here - I was getting pretty weird errors without the first commit though. You can see a pretty minimal example of this here - https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=de08d72c9a23062bf3d224f0d873d27d.

I think the changes to the macro are correct, but I'm not an expert with macro_rules so I definitely could have gotten this wrong. Happy to adjust!

Reply:
This is a known bug in the macro. 0358edeb546dbdc9d2f894c19b54c07884d9b9c8 should fix it.

Reply:
Hm, the first commit in this PR looks simpler than that one, but happy to rebase over that (not sure where that's from -- GitHub isn't showing a PR link).

Reply:
It's from #85830.